Make vehicle folding/unfolding an activity.

irwiss opened the pull request:
#### Summary
None

#### Purpose of change

Make vehicle folding/unfolding an activity.

#### Describe the solution

This allows folding/unfolding to take reasonable times instead of substracting 500 moves.
Also polishes stuff from #59572 a bit, folding parts can specify:
* Tools required; only hand_pump defined for inflatable boat right now, add small wrench for folding frames?
* Time required; 10 seconds per part unless specified, for sanity checking set 1 minute per frames, needs a json pass.

Adds back some of the testing removed in #59572 - tests degradation roundtrips via serialization, and snapshots a few degradation presets and their outcomes.

#### Describe alternatives you've considered

#### Testing

Testing matches #59572 almost exactly, it also has a save prior to the changes attached for testing
Also has some automated tests for the activities themselves

#### Additional context

Fris0uman commented on doc/JSON_INFO.md:
```diff
@@ -2812,6 +2812,8 @@ Vehicle components when installed on a vehicle.
   { "id": "pot" }
 ],
+"fold_tools": [ "needle" ], // tool itype_ids required for folding
+"unfold_tools": [ "hand_pump" ], // tool itype_ids required for unfolding
```
should probably add (un)folding_time too

irwiss commented:
Applied review comments;
* Adjusted boat description
* Fixed up JSON_INFO.md with the new fields
* Added fold/unfold time to inflatable boat parts so it unfolds in ~20 minutes, folds in ~8 minutes

20 minutes is taken from [comment on a boat from amazon](https://www.amazon.com/Intex-Excursion-Inflatable-Boat-Series/dp/B005S6GY74?th=1) and eyeballing from how long it took to inflate an air mattress before electric pump
